libraries/AP_Logger/LoggerMessageWriter.cpp

```diff
@@ -199,6 +211,19 @@ bool LoggerMessageWriter_DFLogStart::writeallrallypoints()
 }
 #endif
 
+#if AC_FENCE
+bool LoggerMessageWriter_DFLogStart::writeallfence()
+{
+    if (stage != Stage::DONE) {
+        return false;
+    }
+    stage = Stage::RUNNING_SUBWRITERS;
```

Comment:
This is weird - this isn't part of that enum.  Looks like we need to sprinkle some `enum class`s around.

Reply:
Its not the same stage enum as the one added in this PR. I classed them all in anycase.